tests/ecosystem/upgrade/test_upgrade.py

```diff
@@ -129,9 +131,29 @@ def test_upgrade():
         resource_name=constants.OPERATOR_CATALOG_SOURCE_NAME,
         namespace=constants.MARKETPLACE_NAMESPACE,
     )
+    version_before_upgrade = config.ENV_DATA.get("ocs_version")
+    upgrade_version = config.UPGRADE.get(
+        "upgrade_ocs_version", version_before_upgrade
+    )
+    version_change = version_before_upgrade != upgrade_version
```

Comment:
Should we ensure that `upgrade_version > version_before_upgrade` here to stop us from trying to "upgrade" to a lower version?

Reply:
+1 on that

Reply:
I added assert in next commit. But need to check with >= cause in case we upgrade from 4.2.1 to 4.2.2 the x.y version is still 4.2 so I don't want to fail on such upgrade.

Reply:
@petr-balogh can we keep track of the full version as well as the standard version we are using now? For example:

```python
ocs_version = '4.2'
full_ocs_version = '4.2.1'

upgrade_version = '4.2'
full_upgrade_version = '4.2.2'
```
If we were able to use the full versions in this check we would be able to use `>` and still support upgrading between minor version increases. If there is a reason why we are limited here then i support sticking with `>=` for now, but I think being able to discern between these minor versions will be a big help going forward. I assume we will need to test upgrade between minor versions quite frequently as more are released.

Reply:
As @clacroix12 suggested, we need to provide precise full_ocs_version for upgrade

Reply:
You can pass 4.2.1 to the --ocs-version parameter or --upgrade-ocs-version parameter but We need to have config file for each z version. So I think that approach I have is more flexible and usually we will upgrade from latest 4.2 to latest 4.3 so between latest X versions. If needed to test specific version than we will create config file wit x.y.z and than we will load it. No need to add any extra full version to the config as internal builds have csv version like 0.0.310 for example. I have opened bz for it today. I will add it to the agenda for automation mtg and we can discuss but I don't want to block this PR on it.

Another disadvantage for keeping full version in config makes impossible to test automatically on new Z build by UMB cause before we will need to update config.

Reply:
And one more thing. It's mainly about reload config for version and we also have configs just for 4.x via --ocs-version . In case we switch between version we just need reload proper config. 

In case you would like to use specific version that's why we have that upgrade-ocs-registry-image where you specify exact image of specific build.

Makes sense @clacroix12 @vavuthu ?

Reply:
I don't think we were necessarily talking about having config files for each z version, but rather having some way of determining the full version that is installed as well as the one we want to upgrade to. This way we can ensure that we aren't trying to upgrade to the same version that is installed.

> I will add it to the agenda for automation mtg and we can discuss but I don't want to block this PR on it.

Agreed, I don't want to either and I think that using `>=` is totally sufficient for now.